File: frontend/lib/modules/annotation/models/response/annotation_file_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'annotation_file_response.g.dart';

@JsonSerializable()
class AnnotationFileResponse {
  final int annotationId;
  final String annotationPath;
  final List<String> files;
  final List<String> classes;
  final int storageType;

  AnnotationFileResponse({
    required this.annotationId,
    required this.annotationPath,
    required this.files,
    required this.classes,
    required this.storageType,
  });

  factory AnnotationFileResponse.fromJson(Map<String, dynamic> json) =>
      _$AnnotationFileResponseFromJson(json);

  Map<String, dynamic> toJson() => _$AnnotationFileResponseToJson(this);
}

File: automl/src/main/java/org/xiaoshuyui/automl/module/annotation/entity/AnnotationFileResponse.java
package org.xiaoshuyui.automl.module.annotation.entity;

import java.util.List;
import lombok.Data;

@Data
public class AnnotationFileResponse {
  long annotationId;
  String annotationPath;
  List<String> files;
  List<String> classes;
  int storageType;
}
